backends/dpdk/dpdkMetadata.cpp

```diff
@@ -35,14 +35,18 @@ void DirectionToRegRead::uniqueNames(IR::DpdkAsmProgram* p) {
 }
 
 const IR::Node* DirectionToRegRead::preorder(IR::DpdkAsmProgram* p) {
+    bool is_direction_used = false;
+    auto IsDirUsed = new IsDirectionMetadataUsed(is_direction_used);
+    IsDirUsed->setCalledBy(this);
+    p->apply(*IsDirUsed);
     uniqueNames(p);
     p->externDeclarations.push_back(addRegDeclInstance(registerInstanceName));
-    IR::IndexedVector<IR::DpdkAsmStatement> stmts;
-    for (auto stmt : p->statements) {
-        auto stmtList = stmt->to<IR::DpdkListStatement>()->clone();
-        stmts.push_back(replaceDirection(stmtList));
+    if (is_direction_used) {
+        IR::IndexedVector<IR::DpdkAsmStatement> stmts;
+        stmts.push_back(new IR::DpdkListStatement(replaceDirectionWithRegRead(
```

Comment:
Name of this function can be changed to insertRegReadForDirection or something similar